src/ert/_c_wrappers/enkf/ecl_config.py

```diff
@@ -1,207 +1,96 @@
-
-            # REFCASE_KEY
-            refcase_default = config_dict.get(ConfigKeys.REFCASE)
-            if refcase_default is not None:
-                refcase_default = os.path.realpath(refcase_default)
-
-            # SCHEDULE_PREDICTION_FILE_KEY
-            schedule_prediction_file = config_dict.get(
-                ConfigKeys.SCHEDULE_PREDICTION_FILE
+        if ConfigKeys.GRID in config_dict:
+            ecl_config_args["grid_file"] = os.path.realpath(
+                config_dict[ConfigKeys.GRID]
             )
-
-            c_ptr = self._alloc_full(
-                have_eclbase,
-                data_file,
-                grid,
-                refcase_default,
-                schedule_prediction_file,
+        if ConfigKeys.REFCASE in config_dict:
+            ecl_config_args["refcase_file"] = os.path.realpath(
+                config_dict[ConfigKeys.REFCASE]
             )
-            if grid is not None:
-                grid.convertToCReference(None)
-
-        if c_ptr:
-            super().__init__(c_ptr)
-        else:
-            raise RuntimeError("Internal error: Failed constructing EclConfig!")
-
-    def free(self):
-        self._free()
-
-    def getDataFile(self):
-        return self._get_data_file()
-
-    def setDataFile(self, datafile):
-        self._set_data_file(datafile)
-
-    def validateDataFile(self, datafile) -> UIReturn:
-        """@rtype: UIReturn"""
-        return self._validate_data_file(datafile)
-
-    # -----------------------------------------------------------------
-
-    def get_gridfile(self) -> str:
-        return self._get_gridfile()
-
-    def set_gridfile(self, gridfile):
-        self._set_gridfile(gridfile)
-
-    def validateGridFile(self, gridfile):
-        return self._validate_gridfile(gridfile)
-
-    def getGrid(self) -> EclGrid:
-        return self._get_grid()
-
-    def getRefcaseName(self):
-        return self._get_refcase_name()
-
-    def loadRefcase(self, refcase):
-        self._load_refcase(refcase)
-
-    def getRefcase(self) -> EclSum:
-        refcase = self._get_refcase()
-        if refcase is not None:
-            refcase.setParent(self)
-
-        return refcase
-
-    def validateRefcase(self, refcase):
-        return self._validate_refcase(refcase)
-
-    def hasRefcase(self):
-        """@rtype: bool"""
-        return self._has_refcase()
-
-    # -----------------------------------------------------------------
-
-    def getDepthUnit(self):
-        return self._get_depth_unit()
-
-    def getPressureUnit(self):
-        return self._get_pressure_unit()
-
-    # -----------------------------------------------------------------
-
-    def active(self):
-        """
-        Has ECLIPSE been configured?"
-        """
-        return self._active()
-
-    def getLastHistoryRestart(self):
-        return self._get_last_history_restart()
+        return cls(**ecl_config_args)
+
+    def _load_grid(grid_file: str) -> Optional[EclGrid]:
```

Comment:
i might not understand this fully, but the assignment of `self._grid_file` in the initialization and then the later `EclConfig._load_grid(grid_file)` .... Why do we need to add the `grid_file` as a parameter? Could the `_load_grid(..)` be called without an argument and inside the function just use the already assigned `self._grid_file`?

Reply:
ja indeed my intention here is that i had in the back of my head that there should be as little logic in the subconfig objects as possible, or at least all parsing should happen in the res config, or something along those lines.

so i wanted to make it clear that this function can basically stand on its own, I guess it doesn't even need to be defined as part of the class, can just be used by it. i'm gonna try to make it so.

Reply:
that sounds strangely reasonable, and like it doesn't even need deep understanding of object oriented stuff :sweat_smile:

ok so simple instance methods, that don't get the files as arguments, but grab them from the instance, got it!